### PR TITLE
ICO content view

### LINKED ISSUE #2407: Add .ico content view
@GreyCat added a kaitai-struct parser for ICO files in https://github.com/kaitai-io/kaitai_struct_formats/commit/22ecfbf7759c3e575bba092b27ddcc1e8f1aff9e. :tada:
Given that favicons are quite common in mitmproxy, adding this would be a fantastic addition to the project!

### PR DESCRIPTION
**Pull request by ujjwal96**
Added ico content view
closes #2407
screenshot from 2017-06-24 03-51-34

### CODE REVIEW

File: mitmproxy/contentviews/image/image_parser.py [hunk 1]
```diff
@@ -78,3 +79,24 @@ def parse_jpeg(data: bytes) -> Metadata:
...
+    img = ico.Ico(KaitaiStream(io.BytesIO(data)))
+    parts = [
+        ('Format', 'ICO'),
+        ('Number of images', str(img.num_images)),
+    ]
+
+    for i, image in enumerate(img.images):
+        parts.append(
+            (
+                'Image {}'.format(i), "Size: {} x {}\n                  "
```
Comment (ujjwal96) [inline mitmproxy/contentviews/image/image_parser.py]: @mhils Any better way to do this? :sweat_smile:   
`\t` won't work in urwid

**Comment by ujjwal96** [comment]
Ready for review

File: mitmproxy/contentviews/image/image_parser.py [hunk 2]
```diff
@@ -78,3 +79,25 @@ def parse_jpeg(data: bytes) -> Metadata:
...
+    parts = [
+        ('Format', 'ICO'),
+        ('Number of images', str(img.num_images)),
+    ]
+
+    for i, image in enumerate(img.images):
+        parts.append(
+            (
+                'Image {}'.format(i + 1), "Size: {} x {}\n"
+                                          "{: >18}Bits per pixel: {}\n"
```
Comment (mhils) [inline mitmproxy/contentviews/image/image_parser.py]: Neat 😄

File: test/mitmproxy/contentviews/image/test_view.py [hunk 3]
```diff
@@ -9,8 +9,7 @@ def test_view_image():
         "mitmproxy/data/image.png",
         "mitmproxy/data/image.gif",
         "mitmproxy/data/all.jpeg",
-        # https://bugs.python.org/issue21574
-        # "mitmproxy/data/image.ico",
+        "mitmproxy/data/image.ico",
```
Comment (mhils) [inline test/mitmproxy/contentviews/image/test_view.py]: The previously linked bug does not apply anymore? If so, this is LGTM!

Comment (ujjwal96) [inline test/mitmproxy/contentviews/image/test_view.py]: Nope, it is about porting image type detections from PIL to imghdr. I add the test for ico to imghdr.tests so we can detect .ico from imghdr.what

Comment (mhils) [inline test/mitmproxy/contentviews/image/test_view.py]: Ah, I see. Makes sense.

**Pull request merged**